### PR TITLE
Upgrade to ClimaTimeStepeprs v0.7.1

### PR DESCRIPTION
**Pull request by valeriabarra**
## Purpose 
In order for us to test the new SSP time-stepper, we need to upgrade to ClimaTimeSteppers v0.7.1

## Content
- [x] This PR upgrades the `perf`, `examples`, and `ode_compat_examples` to use the latest ClimaTimeSteppers v0.7.1
- [x] It renames `IMEXARKAlgorithm` -> `IMEXAlgorithm`
- [x] It renames `AbstractIMEXARKTableau` -> `AbstractAlgorithmName`

- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

**Comment by valeriabarra** [comment]
This will close #1318

**Comment by sriharshakandala** [comment]
The package update part looks good.

File: examples/Project.toml [hunk 1]
```diff
@@ -70,7 +70,7 @@ ClimaCommsMPI = "0.3"
 ClimaCore = "0.10"
 ClimaCorePlots = "0.2"
 ClimaCoreVTK = "0.6, 0.7"
-ClimaTimeSteppers = "0.5, 0.6"
+ClimaTimeSteppers = "0.5, 0.6, 0.7"
```
Comment (dennisYatunin) [inline examples/Project.toml:73]: Should probably remove 0.5 and 0.6, since the API is different for 0.7

File: examples/hybrid/types.jl [hunk 2]
```diff
@@ -284,7 +284,7 @@ import ClimaTimeSteppers as CTS
 import OrdinaryDiffEq as ODE
 
 is_imex_CTS_algo_type(alg_or_tableau) =
-    alg_or_tableau <: CTS.AbstractIMEXARKTableau
+    alg_or_tableau <: CTS.AbstractAlgorithmName
```
Comment (dennisYatunin) [inline examples/hybrid/types.jl:287]: I think this should be an `CTS.IMEXARKAlgorithmName`

File: perf/Project.toml [hunk 3]
```diff
@@ -70,7 +70,7 @@ ArtifactWrappers = "0.2"
 AtmosphericProfilesLibrary = "0.1"
 CLIMAParameters = "0.4, 0.5, 0.6, 0.7"
 ClimaCore = "0.10"
-ClimaTimeSteppers = "0.5, 0.6"
+ClimaTimeSteppers = "0.5, 0.6, 0.7"
```
Comment (dennisYatunin) [inline perf/Project.toml:73]: Also might want to remove 0.5 and 0.6 here

**Review by dennisYatunin — APPROVED** [review]
Looks good overall!

**Pull request merged**